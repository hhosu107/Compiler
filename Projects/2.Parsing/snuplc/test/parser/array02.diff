2d1
< Find subroutine ReadInt
18,19c17,18
<         := <INVALID>
<           [ @A        <array 5  of <array 5  of <int>>> ] <INVALID>
---
>         := <int>
>           [ @A        <array 5  of <array 5  of <int>>> ] <int>
23,24c22,23
<         := <INVALID>
<           [ @A        <array 5  of <array 5  of <int>>> ] <INVALID>
---
>         := <int>
>           [ @A        <array 5  of <array 5  of <int>>> ] <int>
27c26
<           [ @A        <array 5  of <array 5  of <int>>> ] <INVALID>
---
>           [ @A        <array 5  of <array 5  of <int>>> ] <int>
31c30
<           [ @A        <array 5  of <array 5  of <int>>> ] <INVALID>
---
>           [ @A        <array 5  of <array 5  of <int>>> ] <int>
