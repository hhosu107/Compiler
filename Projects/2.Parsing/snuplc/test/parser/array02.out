parsing 'array02.mod'...
Find subroutine ReadInt
successfully parsed.
  AST:
    CAstScope: 'array02'
      symbol table:
        [[
          [ @A        <array 5  of <array 5  of <int>>>           ]
          [ *DIM(<ptr(4) to <NULL>>,<int>) --> <int>           ]
          [ *DOFS(<ptr(4) to <NULL>>) --> <int>           ]
          [ *ReadInt() --> <int>           ]
          [ *WriteChar(<char>) --> <NULL>           ]
          [ *WriteInt(<int>) --> <NULL>           ]
          [ *WriteLn() --> <NULL>           ]
          [ *WriteStr(<ptr(4) to <array  of <char>>>) --> <NULL>           ]
        ]]
      statement list:
        := <INVALID>
          [ @A        <array 5  of <array 5  of <int>>> ] <INVALID>
            2 <int>
            3 <int>
          call [ *ReadInt() --> <int> ] <int>
        := <INVALID>
          [ @A        <array 5  of <array 5  of <int>>> ] <INVALID>
            0 <int>
            1 <int>
          [ @A        <array 5  of <array 5  of <int>>> ] <INVALID>
            2 <int>
            3 <int>
        call [ *WriteInt(<int>) --> <NULL> ] <NULL>
          [ @A        <array 5  of <array 5  of <int>>> ] <INVALID>
            0 <int>
            1 <int>
        call [ *WriteLn() --> <NULL> ] <NULL>
      nested scopes:
        empty.
    


run the following command to convert the .dot file into a PDF:
  dot -Tpdf -oarray02.mod.ast.pdf array02.mod.ast.dot


Done.
